pkg/bdb/scanner.go

```diff
@@ -139,7 +142,9 @@ func NewPageIterator[T any, M Message[T]](ctx context.Context, tx *bolt.Tx, path
 		return nil, err
 	}
 
-	return &PageIterator[T, M]{iter: iter.(*ScanIterator[T, M])}, nil
+	scanIter, _ := iter.(*ScanIterator[T, M])
+
+	return &PageIterator[T, M]{iter: scanIter}, nil
```

Comment:
I'm surprised this needs a runtime type assertion. If `iter` is of type `*ScanIterator[T, M]` then I'd expect it to be possible to simply:
```go
return &PageIterator[T, M]{iter: iter}, nil
```

Reply:
Compiler complains:

```
cannot use iter (variable of interface type Iterator[T, M]) as *ScanIterator[T, M] value in struct literal: need type assertioncompiler[IncompatibleAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#IncompatibleAssign)
var iter Iterator[T, M]
```

Reply:
The problem is here: https://github.com/aserto-dev/go-edge-ds/blob/fe939b6e796d92f9ad277495f2d6decf21fdbb2e/pkg/bdb/scanner.go#L60

`NewScanIterator` should return a `*ScanIterator[T, M]`, not `Iterator[T, M]`.
I'm surprised this isn't caught by the `ireturn` linter.